src/lib/libwebgl.js

```diff
@@ -663,12 +663,24 @@ for (/**@suppress{duplicate}*/var i = 0; i <= {{{ GL_POOL_TEMP_BUFFERS_SIZE }}};
       var contextHandle = glCtx.canvas.GLctxObject.handle;
       glCtx[f] = function(...args) {
         var ret = orig.apply(this, args);
-        // Some GL functions take a view of the entire linear memory.  Replace
-        // such arguments with the string 'HEAP' to avoid serializing all of
-        // memory.
         for (var i in args) {
-          if (ArrayBuffer.isView(args[i]) && args[i].byteLength === HEAPU8.byteLength) {
-            args[i] = 'HEAP';
+          if (ArrayBuffer.isView(args[i])) {
+            // Some GL functions take a view of the entire linear memory.  Replace
+            // such arguments with the string 'HEAP' to avoid serializing all of
+            // memory.
+            if (args[i].byteLength === HEAPU8.byteLength) {
```

Comment:
This could have compared `.buffer` between them to be more precise, but this is probably good enough in practice.

Reply:
Just FYI, this is pre-existing code